rustls/src/client/hs.rs

```diff
@@ -271,6 +271,47 @@ fn emit_client_hello_for_retry(
     // Do we have a SessionID or ticket cached for this host?
     let tls13_session = prepare_resumption(&input.resuming, &mut exts, suite, cx, config);
 
+    // Extensions MAY be randomized
+    // but they also need to keep the same order as the the previous ClientHello
+    exts.sort_by_cached_key(|new_ext| {
+        match new_ext {
+            // PresharedKey is always be last
+            ClientExtension::PresharedKey(..) => usize::max_value(),
```

Comment:
I'd propose the following structure:

```rust
if let ClientExtension::PresharedKey(..) = new_ext {
    return usize::MAX;
}

if input.hello.sent_extensions.is_empty() {
     ..
}

..
```

This helps reduce rightward drift. (And replaces `max_value()` with `MAX`, which is more idiomatic.)